src/main/java/bio/terra/service/cohortbuilder/InstancesService.java

```diff
@@ -76,58 +87,213 @@ public InstanceCountList countInstances(String entityName, CountQuery countQuery
   }
 
   public InstanceList queryInstances(String entityName, Query query) {
-    List<InstanceRelationshipFields> relationshipFields =
-        List.of(
+    ValidationUtils.validateApiFilter(query.getFilter());
```

Comment:
This code is copy pasted from the InstanceController in tanagra with a few intellij style changes